Lib/logging/handlers.py

```diff
@@ -827,11 +828,26 @@ def __init__(self, address=('localhost', SYSLOG_UDP_PORT),
             self.unixsocket = False
             if socktype is None:
                 socktype = socket.SOCK_DGRAM
-            self.socket = socket.socket(socket.AF_INET, socktype)
-            if socktype == socket.SOCK_STREAM:
-                self.socket.connect(address)
-            self.socktype = socktype
-        self.formatter = None
+            host, port = address
+            err = None
+            for res in socket.getaddrinfo(host, port, 0, socktype):
+                af, socktype, proto, cannonname, sa = res
+                sock = None
+                try:
+                    sock = socket.socket(af, socktype, proto)
+                    if socktype == socket.SOCK_STREAM:
+                        sock.connect(sa)
+                    self.socket = sock
+                    self.socktype = socktype
+                    return
+                except OSError as _:
+                    err = _
+                    if sock is not None:
+                        sock.close()
+            if err is not None:
+                raise err
+            else:
```

Comment:
Fair enough. But we can't make `else` belong to the `for` loop here. If all the results are tried and fail, we could also go into the `else` branch and give wrong message. I reformat the code a bit, please see it's okay or not. :-)